test/integration-test.js

```diff
@@ -592,6 +604,52 @@ describe('Integration Test', () => {
         });
       });
     });
+
+    it('needsFlattenedData', () => {
+      const appDef = _.cloneDeep(appDefinition);
+      appDef.legacy.needsFlattenedData = true;
+      const _appDefWithAuth = withAuth(appDef, apiKeyAuth);
+      const _compiledApp = schemaTools.prepareApp(_appDefWithAuth);
+      const input = createTestInput(
+        _compiledApp,
+        'triggers.movie.operation.perform'
+      );
+      input.bundle.authData = { api_key: 'secret' };
+      input.bundle.meta = {
+        zap: { name: 'My Awesome Zap' }
+      };
+      return app(input).then(output => {
+        should.equal(typeof output.results, 'string');
```

Comment:
Does the behavior in this test match the Python backend? Seeing this makes me think we might need to make some adjustments. I think we still want an array back, but any lists within the individual elements should be flattened to a csv string.